### DISCUSSION TITLE
The curious case of Dalvik, two IPs and the apple-touch-icon.png request

### DISCUSSION THREAD

**Discussion by 737simpilot** [CIDRAM/CIDRAM#298] [Q&A]
I'm seeing traffic from what looks like a legit user using a mobile IP from AT&T and a device of Windows 10 using the latest version of Chrome. But I'm also seeing the same user using another IP from AT&T at the same time but with a UA of: Dalvik/2.1.0 (Linux; U; Android 11; DE2118 Build/RKQ1.201217.002) and another UA of: Mozilla/5.0 (Linux; Android 11) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/99.0.4844.73 Mobile DuckDuckGo/5 Safari/537.36

So this user is using three UAs and two IPs. The interesting part is the 404s requesting apple-touch-icon.png from the Dalvik UA and only the Dalvik UA. From my reading this request for the apple-touch-icon.png is common for Apple products. And if you don't satisfy that request you'll see 404s in your log for that request. 

Now, while it seems like legit traffic on its face from the first UA and IP, I'm not understanding why there's three UAs and two IPs? From a cell carrier?  Especially since none of these UAs indicate an Apple device coinciding to the request for the apple-touch-icon.png.

This is the UA that looked like legit traffic at the website:

Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36

Here are the two UAs used with another IP from the same user. (verified it's the same user based on time in logs and requests):

Mozilla/5.0 (Linux; Android 11) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/99.0.4844.73 Mobile DuckDuckGo/5 Safari/537.36

Dalvik/2.1.0 (Linux; U; Android 11; DE2118 Build/RKQ1.201217.002)

This appears to be the phone: `https://devicebeast.com/devices/oneplus-nord-n200-5g-td-lte-us-64gb-de2118`

And here's the OS this thing runs: `https://en.wikipedia.org/wiki/OxygenOS`

So, perhaps I just answered my own question? The OS is to blame because it's pulling shenanigans? But two IPs? Why all the UAs for god sakes? Doesn't appear this phone can be flashed with a third party OS like LineageOS. Or at least not very easily. 

My research:

https://stackoverflow.com/questions/12480497/why-am-i-getting-error-for-apple-touch-icon-precomposed-png

https://stackoverflow.com/questions/23804278/browser-sending-dalvik-as-user-agent

https://en.wikipedia.org/wiki/Dalvik_(software)

**Comment by 737simpilot** [comment]
= Useless information = 

AT&T started out in 1885 (same year as in Back To The Future III). The acronym is the American Telephone and Telegraph company formed by Alexander Gram Bell. 

They provide landline service as well as cellphone service. And back in the day, I used them for dial-up. They probably own a crap ton of satellites among other infrastructure like microwave and fiber.

Google searches will yield NSA shenanigans, but there's CALEA so all telephone and Internet companies have to comply. 

https://en.wikipedia.org/wiki/Communications_Assistance_for_Law_Enforcement_Act

**Comment by Maikuolan** [comment]
*Related: #348.*
